shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/BlockChain.hs

```diff
@@ -637,23 +642,19 @@ checkLeaderValue certVRF σ f =
     -- This is a testing convenience, the active slot coefficient should not
     -- bet set above one half otherwise.
       True
-    else
-      if leaderVal == 1
-        then -- Having a VRF value of one is always a failure.
-        -- Moreover, it would cause division by zero.
-          False
-        else case taylorExpCmp 3 (1 / q) x of
-          ABOVE _ _ -> False
-          BELOW _ _ -> True
-          MaxReached _ -> False
+    else case taylorExpCmp 3 q x of
+      ABOVE _ _ -> False
+      BELOW _ _ -> True
+      MaxReached _ -> False
   where
-    leaderVal = (unitIntervalToRational . fromNatural) certNat
+    certNatMax :: Natural
+    certNatMax = (2 :: Natural) ^ (8 * VRF.sizeOutputVRF (Proxy @v))
+    c, q, x :: FixedPoint
     c = activeSlotLog f
-    q = fromRational $ 1 - leaderVal
+    q = fromRational (toInteger certNatMax % toInteger (certNatMax - certNat))
```

Comment:
So this is now still not in line with the floating point doc right? (Meaning of `q` is still different.)

Reply:
Can you point out where `q` is defined in the non-integer documentation? I don't see what you are referring to (looking at page 111 of https://hydra.iohk.io/build/3156550/download/1/ledger-spec.pdf)

Reply:
https://hydra.iohk.io/build/3156326/download/1/non-integer-calculations.pdf , section 6,  page 10.

Reply:
Aha! For some reason I thought that the existence of the `non-integer` sectoin in the ledger spec meant that entire document had been merged. It did seem rather short...